sw/device/lib/drivers/fast_intr_ctrl/fast_intr_ctrl.c

```diff
@@ -1,24 +1,431 @@
+ * @brief Fast gpio 4 irq handler. The first entry point when gpio 4 interrupt
+ * is recieved through fic.
+ * This function clear the responsible bit in FAST_INTR_PENDING then call a 
+ * function that can be overriden inside peripherals.
+ */
+INTERRUPT_HANDLER_ABI void handler_irq_fast_gpio_4(void);
+
+/**
+ * @brief Fast gpio 5 irq handler. The first entry point when gpio 5 interrupt
+ * is recieved through fic.
+ * This function clear the responsible bit in FAST_INTR_PENDING then call a 
+ * function that can be overriden inside peripherals.
+ */
+INTERRUPT_HANDLER_ABI void handler_irq_fast_gpio_5(void);
+
+/**
+ * @brief Fast gpio 6 irq handler. The first entry point when gpio 6 interrupt
+ * is recieved through fic.
+ * This function clear the responsible bit in FAST_INTR_PENDING then call a 
+ * function that can be overriden inside peripherals.
+ */
+INTERRUPT_HANDLER_ABI void handler_irq_fast_gpio_6(void);
+
+/**
+ * @brief Fast gpio 7 irq handler. The first entry point when gpio 7 interrupt
+ * is recieved through fic.
+ * This function clear the responsible bit in FAST_INTR_PENDING then call a 
+ * function that can be overriden inside peripherals.
+ */
+INTERRUPT_HANDLER_ABI void handler_irq_fast_gpio_7(void);
+
+/****************************************************************************/
+/**                                                                        **/
+/*                           EXPORTED VARIABLES                             */
+/**                                                                        **/
+/****************************************************************************/
+
+/****************************************************************************/
+/**                                                                        **/
+/*                            GLOBAL VARIABLES                              */
+/**                                                                        **/
+/****************************************************************************/
+
+/****************************************************************************/
+/**                                                                        **/
+/*                           EXPORTED FUNCTIONS                             */
+/**                                                                        **/
+/****************************************************************************/
+
+fast_intr_ctrl_result_t clear_fast_interrupt(fast_intr_ctrl_fast_interrupt_t\
+ fast_interrupt)
 {
+    /* masking a 32b go write in reg*/
     uint32_t reg = 0;
-
     reg = bitfield_bit32_write(reg, fast_interrupt, true);
-    mmio_region_write32(fast_intr_ctrl->base_addr, (ptrdiff_t)(FAST_INTR_CTRL_FAST_INTR_CLEAR_REG_OFFSET), reg);
-
+    /* write in reg through structure */
+    fast_intr_ctrl_peri->FAST_INTR_CLEAR = reg;
     return kFastIntrCtrlOk_e;
 }
+
+__attribute__((weak)) void fic_irq_timer_1(void)
+{
+    volatile uint8_t i;
```

Comment:
remove this, add:
`/* Users should implement their non-weak version */`

Reply:
It is not working in this way, if I replace it strong function never calls

Reply:
yes, you have to change the 
`__attribute__((weak))`
to
`__attribute__((weak, optimize("O0")))`
Note that we are compiling the whole thing with '0s', i.e. all optimisations till '02' are active. If we specify just '__weak__' dead-code optimisation is applied over it

Reply:
DONE